src/olympia/blocklist/views.py

```diff
@@ -111,15 +125,25 @@ def get_items(apiver, app, appver=None):
     return items, details
 
 
-def get_plugins(apiver, app, appver=None):
+def get_plugins(apiver=3, app=None, appver=None):
     # API versions < 3 ignore targetApplication entries for plugins so only
     # block the plugin if the appver is within the block range.
+
+    if app:
+        app_query = (Q(app__isnull=True) |
+                     Q(app__guid=app) |
+                     Q(app__guid__isnull=True))
+    else:
```

Comment:
Can't find a better solution than you. I think keeping `Q` here is fine, since it's already what was done, and the goal is to get rid of all this mess anyway.

I suggest the following though:
- Change `app_query` to `Q(app__isnull=True) | Q(app__isnull=True)`. Assuming I understood correctly, you only need the `Q` filtering to force the `LEFT OUTER JOIN`, and it seems to be enough to do the trick. Do double check :)
- Speaking of double-checking, add a test that would fail if the join was done improperly ?
- Add a comment explaining why this `Q` is useful, and a docstring to `_blocklist_json` to explain what it's trying to achieve by calling with `app=None`. This is especially important as it differs from the original `_blocklist`.